react-native/index.d.ts

```diff
@@ -4554,12 +4554,22 @@ declare module "react" {
     // @see https://github.com/facebook/react-native/blob/0.34-stable\Libraries\StyleSheet\StyleSheetTypes.js
     export namespace StyleSheet {
 
+        interface Styles {
+            [style: string]: Style
+        }
+
         type Style = ViewStyle | TextStyle | ImageStyle
 
         /**
          * Creates a StyleSheet style reference from the given object.
          */
-        export function create<T>( styles: T ): T;
+        // Non-generic override is required to provide intellisense
+        // for JavaScript and non-generic method invocations
+        export function create(styles: Styles): any;
```

Comment:
@vladimir-kotikov  I don't quite see it. Your example works just fine without this change because the generic parameter T contains the foo property. Could you perhaps add a test that would fail without your change?

As this PR is getting stale I plan to close it if no response within a few days.